### PR TITLE
[gh-#1216] add google pagination and gcal syncs

### PR DESCRIPTION
**Pull request by khaliqgant**
Adds pagination and proxy for google and adds a google-calendars and google-calendar-events sync

### CODE REVIEW

File: integration-templates/google-calendar/google-calendar-events.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+            if (batch.length === batchSize) {
+                await nango.batchSave<GoogleCalendarEvent>(batch, 'GoogleCalendarEvent');
+            }
+        }
+    }
+
+    if (batch.length > 0) {
+        await nango.batchSave<GoogleCalendarEvent>(batch, 'GoogleCalendarEvent');
+    }
```
Comment (TBonnin) [inline integration-templates/google-calendar/google-calendar-events.ts:15]: why do we need the intermediary `batch` array. can we not batchSave the result of paginate directly?

```
for await (const eventPage of nango.paginate<GoogleCalendarEvent>({ endpoint, params: { maxResults } })) {
   await nango.batchSave<GoogleCalendarEvent>(eventPage, 'GoogleCalendarEvent');
}
```

File: integration-templates/google-calendar/google-calendar-events.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import type { NangoSync, GoogleCalendarEvent } from './models';
+
+export default async function fetchData(nango: NangoSync): Promise<void> {
+    const oneMonthAgo = new Date();
+    oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1);
```
Comment (TBonnin) [inline integration-templates/google-calendar/google-calendar-events.ts:5]: I am curious: why one month?

Comment (khaliqgant) [reply integration-templates/google-calendar/google-calendar-events.ts:5]: Good question -- this was a sync built out for a customer and this was their constraint.

File: integration-templates/google-calendar/google-calendar-events.ts (hunk 1 above)
Comment (khaliqgant) [reply integration-templates/google-calendar/google-calendar-events.ts:15]: good point! 

This was logic copied over from other sync scripts where there is some processing of the data, but in this case thats's not needed. Updated with https://github.com/NangoHQ/nango/pull/1269/commits/f109e782d52f15298e2967223aa076c5d7965b01

**Pull request merged**